### ISSUE TITLE
converting microns to wave number

### ISSUE THREAD

**Issue by alexmaragko** [radio-astro-tools/spectral-cube#881]
I have a spectral cube object where the spectral_axis is in microns (u.um) and I would like to update it, or alternatively create a new cube, in wave numbers (1/u.cm).

new_cube = cube.with_spectral_unit(1/u.cm) does not work because it requires an equivalencies=u.spectral(), but with_spectral_unit takes the velocity_convention argument with the ‘relativistic’, ‘radio’, or ‘optical’ options.

cube.spectral_axis.to(1/u.cm, equivalencies=u.spectral()) does the job but I am not sure how to create a new cube that has this as the spectral axis.

**Comment by keflavich** [comment]
The`velocity_convention` keyword should be optional.  `cube.with_spectral_unit` should then just work.  All conversions of velocity coordinates automatically use the `.spectral` equivalency.

If you get an error leaving `velocity_convention` unspecified - that's a legit bug that we should fix, but put in `velocity_convention='radio'` as a placeholder

**Comment by alexmaragko** [comment]
Giving `nu_cube = scube.with_spectral_unit(1/u.cm)` returns `*** KeyError: 'wavenumber'`, and the same error is given even if I include `velocity_convention='radio'`

**Comment by keflavich** [comment]
Ah, then this is a deeper issue; it may be that the FITS standard does not support wavenumber?  I'll have to dig more deeply to be sure

**Comment by alexmaragko** [comment]
Thanks. Is there currently a way to create a new cube using `cube.spectral_axis.to(1/u.cm, equivalencies=u.spectral())` which works as expected?

**Comment by keflavich** [comment]
No, the framework underlying all of what spectral-cube does is the FITS-WCS library.  If FITS doesn't support a unit, we can't serialize the data or compute transformations b/w systems

**Comment by keflavich** [comment]
Ah, but FITS does support wavenumber; this might be a simple fix
